backend/asmgen.ml

```diff
@@ -279,18 +279,20 @@ let compile_fundecl ~ppf_dump ~funcnames fd_cmm =
   ++ Compiler_hooks.execute_and_pipe Compiler_hooks.Mach_polling
   ++ Profile.record ~accumulate:true "checkmach"
        (Checkmach.fundecl ~future_funcnames:funcnames ppf_dump)
-  ++ Profile.record ~accumulate:true "comballoc" Comballoc.fundecl
-  ++ Compiler_hooks.execute_and_pipe Compiler_hooks.Mach_combine
-  ++ pass_dump_if ppf_dump dump_combine "After allocation combining"
-  ++  (fun fd ->
+  ++ (fun fd ->
       match !Flambda_backend_flags.cfg_cse_optimize with
       | false ->
         fd
+        ++ pass_dump_if ppf_dump dump_combine "Before allocation combining"
+        ++ Profile.record ~accumulate:true "comballoc" Comballoc.fundecl
+        ++ Compiler_hooks.execute_and_pipe Compiler_hooks.Mach_combine
+        ++ pass_dump_if ppf_dump dump_combine "After allocation combining"
         ++ Profile.record ~accumulate:true "cse" CSE.fundecl
         ++ Compiler_hooks.execute_and_pipe Compiler_hooks.Mach_cse
         ++ pass_dump_if ppf_dump dump_cse "After CSE"
-     | true ->
-       fd)
+      | true ->
+        (* Will happen after `Cfgize`. *)
```

Comment:
Is there a check that if `cfg_cse_optimize` is `true` than `regalloc` is not `Upstream`?

Reply:
I was indeed wondering whether we wanted
to error out in such a case.

Reply:
I don't see any reason not to, unless you want this combination for testing/benchmarking for now... though I can't see how it would be useful even for that.

Reply:
That provides a way to disable the
optimization on the mach pipeline,
so that can be used to get some
kind of baseline but indeed outside
of debugging it does not feel
particularly useful. Updated to
error out.